datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -504,6 +530,112 @@ def create_aws_auth(config):
 }
 
 
+def create_auth_token_handler(config):
+    if not isinstance(config, dict):
+        raise ConfigurationError('The `auth_token` field must be a mapping')
+    elif 'reader' not in config or 'writer' not in config:
+        raise ConfigurationError('The `auth_token` field must define both `reader` and `writer` settings')
+
+    config = deepcopy(config)
+
+    reader_config = config['reader']
+    if not isinstance(reader_config, dict):
+        raise ConfigurationError('The `reader` settings of field `auth_token` must be a mapping')
+
+    writer_config = config['writer']
+    if not isinstance(writer_config, dict):
+        raise ConfigurationError('The `writer` settings of field `auth_token` must be a mapping')
+
+    reader_type = reader_config.pop('type', '')
+    if not isinstance(reader_type, str):
+        raise ConfigurationError('The reader `type` of field `auth_token` must be a string')
+    elif not reader_type:
+        raise ConfigurationError('The reader `type` of field `auth_token` is required')
+    elif reader_type not in AUTH_TOKEN_READERS:
+        raise ConfigurationError(
+            'Unknown `auth_token` reader type, must be one of: {}'.format(', '.join(sorted(AUTH_TOKEN_READERS)))
+        )
+
+    writer_type = writer_config.pop('type', '')
+    if not isinstance(writer_type, str):
+        raise ConfigurationError('The writer `type` of field `auth_token` must be a string')
+    elif not writer_type:
+        raise ConfigurationError('The writer `type` of field `auth_token` is required')
+    elif writer_type not in AUTH_TOKEN_WRITERS:
+        raise ConfigurationError(
+            'Unknown `auth_token` writer type, must be one of: {}'.format(', '.join(sorted(AUTH_TOKEN_WRITERS)))
+        )
+
+    reader = AUTH_TOKEN_READERS[reader_type](reader_config)
+    writer = AUTH_TOKEN_WRITERS[writer_type](writer_config)
+
+    return AuthTokenHandler(reader, writer)
+
+
+class AuthTokenHandler(object):
+    def __init__(self, reader, writer):
+        self.reader = reader
+        self.writer = writer
+
+    def poll(self, **request):
+        token = self.reader.read(**request)
+        if token is not None:
+            self.writer.write(token, **request)
+
+
+class AuthTokenFileReader(object):
+    def __init__(self, config):
+        self._path = config.get('path', '')
+        if not isinstance(self._path, str):
+            raise ConfigurationError('The `path` setting of `auth_token` reader must be a string')
+        elif not self._path:
+            raise ConfigurationError('The `path` setting of `auth_token` reader is required')
+
+        # Cache all updates just in case
+        self._token = None
+
+    def read(self, **request):
+        if self._token is None or 'error' in request:
+            with open(self._path, 'rb') as f:
+                self._token = f.read().decode('utf-8').strip()
+
+            return self._token
```

Comment:
```suggestion
        return self._token
```
?

Currently, none is returned is `self._token` already exist.

Reply:
Yes, the handler only writes if the reader indicates that the token has been refreshed:

```
    def poll(self, **request):
        token = self.reader.read(**request)
        if token is not None:
            self.writer.write(token, **request)
```